packages/bsky/src/services/indexing/index.ts

```diff
@@ -93,6 +103,57 @@ export class IndexingService {
     }
   }
 
+  async indexRepo(did: string, commit: string) {
+    const now = new Date().toISOString()
+    const { pds, signingKey } = await this.didResolver.resolveAtpData(did)
+    const { sync } = new AtpAgent({ service: pds }).api.com.atproto
+
+    const { data: car } = await sync.getCheckout({ did, commit })
+    const { root, blocks } = await readCarWithRoot(new Uint8Array(car))
+    const storage = new MemoryBlockstore(blocks)
+    const repo = await verifyCheckoutToRepo(storage, root, did, signingKey)
+
+    // First, replace the user with just their current profile if it exists
+    await this.db.transaction(async (tx) => {
+      const indexingTx = new IndexingService(tx, this.didResolver)
+      await indexingTx.deleteForUser(did)
```

Comment:
I wonder if it could be nice to diff the current indexed records against the new ones so that we're not doing a mass delete + mass add 🤔 

this will also update all the `indexedAt` times for every record - even if it wasn't actually mutated during this change

but i can see the headaches & data load of that back and forth as well 🤔

Reply:
I expect this to get more sophisticated over time, but this should also be rare/discouraged so I figured I'd just go with the simplest implementation to start.  In the worst case scenarios that a repo was truncated or totally replaced this will perform the best, it's the awkward cases where someone actually performed 500 ops that this will be suboptimal.

Re: the `indexedAt` times, it's a good point and I observed this when testing it out.  IMO it's acceptable, aside from the fact that we often sort on `indexedAt`.  My plan is to implement Paul's strategy in the indices, keeping around a sort field based on the combination of of `createdAt` and `indexedAt`.

Reply:
kk cool that should be fine 👌 

yup this will just need to go along with a bit of extra logic around times 👍